packages/widgets/src/indexer-manager/component.tsx

```diff
@@ -0,0 +1,72 @@
+import { useMemo } from "react";
+import { Anchor, Button, Card, Flex, Group, ScrollArea, Text } from "@mantine/core";
+import { useDisclosure } from "@mantine/hooks";
+import { IconCircleCheck, IconCircleX, IconReportSearch, IconTestPipe } from "@tabler/icons-react";
+
+import { clientApi } from "@homarr/api/client";
+import { useI18n } from "@homarr/translation/client";
+
+import type { WidgetComponentProps } from "../definition";
+import { NoIntegrationSelectedError } from "../errors";
+
+export default function IndexerManagerWidget({
+  options,
+  integrationIds,
+  serverData,
+}: WidgetComponentProps<"indexerManager">) {
+  if (integrationIds.length === 0) {
+    throw new NoIntegrationSelectedError();
+  }
+  const t = useI18n();
+  const data = useMemo(() => (serverData?.initialData ?? []).flatMap((indexers) => indexers.indexers), [serverData]);
+
+  const [loading, { toggle }] = useDisclosure();
+  const { mutate: testAll } = clientApi.widget.indexerManager.testAllIndexers.useMutation({
+    onMutate: () => {
+      toggle();
+    },
+    onSettled: () => {
```

Comment:
Just as a question, doesn't this trigger always? So maybe the second one is not required?

Reply:
It's toggle on / off so if I remove onSettled it's stay loading